app/src/utils/addActivity.ts

```diff
@@ -197,12 +203,55 @@ export function populateJurisdictionArray(record) {
   };
 }
 
-export function findSpeciesCodesAndConcatenateLabels(
-  obj: any,
-  targetKey: string,
-  properties: Record<string, any>,
+/**
+ * Works with `findSpeciesCodes()` to get the 'obj' input and processes it to concatenate codes
+ *
+ * @param {Record<string, Set<string>>} obj - The object that contains sets of species codes
+ * @returns {string} A string representation of the concatentaed codes, comma separated
+ */
+export function getConcatenatedCodes(obj: Record<string, Set<string>>): string {
+  return (
+    Object.values(obj)
+      .flatMap((set) => Array.from(set))
+      .join(', ') || ''
+  );
+}
+
+/**
+ * Works with `findSpeciesCodes()` to get the 'obj' input and processes it to extract species' labels
+ *
+ * @param {Record<string, Set<string>>} obj - The object that contains sets of species codes
+ * @param {OfflineDocsProperties} properties - Properties to search from, to map codes to their corresponding labels
+ * @param {boolean} specialCase - A flag for special cases where the code search needs to be altered based on the keys to find
+ * @returns {string} A string representation of the concatenated labels, comma separated
+ */
+export function getConcatenatedLabels(
+  obj: Record<string, Set<string>>,
+  properties: OfflineDocsProperties,
   specialCase: boolean
 ): string {
+  let labels: string[] = [];
+  const keysToFind = ['invasive_plant_code', 'invasive_plant_aquatic_code'];
+  for (const key of keysToFind) {
+    let propertyKey = specialCase ? 'invasive_plant_aquatic_code' : key;
```

Comment:
```suggestion
  const labels: string[] = [];
  const keysToFind = ['invasive_plant_code', 'invasive_plant_aquatic_code'];
  for (const key of keysToFind) {
    const propertyKey = specialCase ? 'invasive_plant_aquatic_code' : key;
```
Neither variables get reassigned and can be kept as constants